test/cases/coerced_tests.rb

```diff
@@ -1667,3 +1686,13 @@ def marshal_fixture_path(file_name)
     )
   end
 end
+
+class PrimaryKeyAnyTypeTest < ActiveRecord::TestCase
+  # Same as original but add 'default: nil' option
+  coerce_tests! %r{schema dump primary key includes type and options}
+  test "schema dump primary key includes type and options coerced" do
+    schema = dump_table_schema "barcodes"
+    assert_match %r/create_table "barcodes", primary_key: "code", id: { type: :string, limit: 42, default: nil }/, schema
```

Comment:
Do you know why we differ from the original test?

Reply:
The difference is due to 
https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/blob/main/lib/active_record/connection_adapters/sqlserver/schema_dumper.rb#L18.

That method was introduced by https://github.com/rails-sqlserver/activerecord-sqlserver-adapter/commit/89ccbe40c611128d9524a11054001eb431c2c1f0.

Now I'm thinking that maybe instead of coercing we should change that condition. I'm not sure what's the best here.

Here are the definitions of other adapters:

[MySQL](https://github.com/rails/rails/blob/v6.1.0/activerecord/lib/active_record/connection_adapters/mysql/schema_dumper.rb#L37)
```
column.type == :integer && !column.auto_increment?
```
[Postgresql](https://github.com/rails/rails/blob/v6.1.0/activerecord/lib/active_record/connection_adapters/postgresql/schema_dumper.rb)
```
column.type == :uuid || (column.type == :integer && !column.serial?)
```

[Sqlite3](https://github.com/rails/rails/blob/v6.1.0/activerecord/lib/active_record/connection_adapters/sqlite3/schema_dumper.rb)
```
column.bigint?
```

By checking `primary_key?` and not ` column.type` string primary keys end up having the explicit default.

By the way, I split this PR into two.

Reply:
I did that. Also made small changes not related to the schema difference but related to primary_key handling:

1. Remove one condition from `default_primary_key?` (redundant from my point of view). 
2. Change `primary_key_non_clustered` type to bigint
3. Remove unnecessary option set to `uuid` primary_keys

Sorry if they were out of scope.

I added a test similar to [the one rails](https://github.com/rails/rails/blob/v6.1.0/activerecord/test/cases/primary_keys_test.rb#L419) has to show that:
1. bigint and integer primary_keys have `:integer` type (I trust on that in `explicit_primary_key_default?`)
2. bigint identity column is the default primary key and not dumped to schema.